template.yml

```diff
@@ -180,13 +180,15 @@ Resources:
                   yes | diskutil repairDisk $PDISK
                   diskutil apfs resizeContainer $APFSCONT 0
                   systemsetup -setcomputersleep never
-                  user=ec2-user
-                  su - "${!user}" -c 'brew install buildkite/buildkite/buildkite-agent'
+                  sudo -u ec2-user -i <<'EOF'
+                  export PATH="/opt/homebrew/bin:$PATH"
```

Comment:
do AWS offer non-ARM mac instances? if so, i thiiiiink that this brew prefix might be wrong, i think `/opt/homebrew/bin` is ARM only and x86 instances of brew use `/usr/local` (which i guess would be on PATH anyway?

Reply:
I think Homebrew install sticks the folllowing into `~/.zprofile`:

```shell
# Set PATH, MANPATH, etc., for Homebrew.
eval "$(/opt/homebrew/bin/brew shellenv)"
```

Maybe this script should do the same?

For posterity here's what `brew shellenv` prints on my machine:

```shell
export HOMEBREW_PREFIX="/opt/homebrew";
export HOMEBREW_CELLAR="/opt/homebrew/Cellar";
export HOMEBREW_REPOSITORY="/opt/homebrew";
export PATH="/opt/homebrew/bin:/opt/homebrew/sbin${PATH+:$PATH}";
export MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";
export INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";
```

Reply:
Good points, I forgot that `/opt/homebrew` was an ARM-ism.

I've replaced the direct manipulation with a call to `brew shellenv` if it's executable.